hawkular-alerts-api/src/main/java/org/hawkular/alerts/api/model/event/Alert.java

```diff
@@ -63,6 +63,9 @@
     private List<Note> notes;
 
     @JsonInclude(Include.NON_EMPTY)
+    private List<LifeCycle> lifecycle;
```

Comment:
> I'm not aware of this being valid or at all common. This sort of lifecycle support has not been a goal for us, and actually, I believe there is code to actively prevent going back to Ack from Resolved.
This was valid from the original Hawkular UI usecases. It was discussed in UX + BU meetings and we took it as valid. 
No, we don't prevent going back manually.
But with the MiQ intergation this logic was deferred in a less priority status.

> Yes, we would have to index the fields but we already have that pattern for a variety of fields, no?
Yes, but the pattern in place is not enough when the ack or resolved time fields may change. There is a need to "delete the last one" when maintaining the auxiliar indexes columns group.

> If we are to add Lifecycle then I wonder the same as @moolitayer, why keep the explicit ack/resolve fields?
This is a valid concern. As we introduced these fields, I don't want to delete them from the json object yet. Also there are some changes on the actions architecture that are related to this change.
So I think is a better/safer approximation to introduce these changes that basically affects to the filtering and remove those changes in a following iteration.